pallets/msa/src/lib.rs

```diff
@@ -1246,14 +1237,42 @@ impl<T: Config + Send + Sync> CheckFreeExtrinsicUse<T> {
 			.into();
 		return ValidTransaction::with_tag_prefix(TAG_PREFIX).and_provides(account_id).build()
 	}
+
+	/// validates that a MSA being retired is valid
+	pub fn validate_msa_retirement(account_id: &T::AccountId) -> TransactionValidity {
+		const TAG_PREFIX: &str = "MSARetirement";
+		let msa_id = Pallet::<T>::ensure_valid_msa_key(account_id)
+			.map_err(|_| InvalidTransaction::Custom(ValidityError::InvalidMsaKey as u8))?
+			.into();
+
+		// Invalid transaction error "InvalidRegisteredProviderCannotBeRetired" if the MSA id is a registered provider
+		ensure!(
+			!Pallet::<T>::is_registered_provider(msa_id),
+			InvalidTransaction::Custom(
+				ValidityError::InvalidRegisteredProviderCannotBeRetired as u8
+			)
+		);
+
+		// Invalid transaction error "MoreThanOneKeyExists" if the MSA has more than one account key.
+		let key_count = Pallet::<T>::get_public_key_count_by_msa_id(msa_id);
+		ensure!(
+			key_count == 1,
+			InvalidTransaction::Custom(ValidityError::InvalidMoreThanOneKeyExists as u8)
+		);
+		return ValidTransaction::with_tag_prefix(TAG_PREFIX).and_provides(account_id).build()
```

Comment:
I don't think you need the return statement.

Reply:
Removed return.